Improve ISSUE_TEMPLATE.md

fmachado091 opened the pull request:
- Fixes typos
- Organizes sections
- Improves formatting and word phrasing
- Removes ADR reference in favor of RFC

Link to the file: [Current](https://github.com/quintoandar/.github/blob/69b5382750ffd799de863133bbc2736d03eb62c7/.github/ISSUE_TEMPLATE.md) vs [Proposed](https://github.com/quintoandar/.github/blob/35c8b3ccdcd9c5d7b167533dfdce03b82f4369b2/.github/ISSUE_TEMPLATE.md)

fmachado091 committed:
Improve ISSUE_TEMPLATE.md

- Fixes typos
- Organizes sections
- Improves formatting and word phrasing
- Removes ADR reference in favor of RFC

lucianohgo reviewed (approved):
Cool :) Might be good to have somewhere a way to add what jira ticket this is related to. Could be optional since not all teams use Jira

danielasy commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -4,71 +4,56 @@ labels: 'tech-debt'
-Here you should add only hints, context, and opinions
-but let the solution proposal in a PR for ADR.
--->
+Include a description of the current or possible impact of this tech debt and when it's estimated to become critical.
```
Maybe we can keep the new hint texts as comments, just like the label explanation at the top, so people won't have to delete it when filling in the sections

```suggestion
<!--
Include a description of the current or possible impact of this tech debt
and when it's estimated to become critical.
-->
```

woliveiras reviewed (approved):
![Lança!](https://media.giphy.com/media/tXLpxypfSXvUc/giphy.gif)

fmachado091 commented:
@lucianohgo 

> Cool :) Might be good to have somewhere a way to add what jira ticket this is related to. Could be optional since not all teams use Jira

Do you think it could be in the "further info" section?

fmachado091 committed:
Update .github/ISSUE_TEMPLATE.md

Co-authored-by: Daniela Yassuda <danielayassuda@gmail.com>

fmachado091 committed:
Make more improvements

- Turn placeholder texts into comments;
- Add JIRA reference to the "Further info" section;
- Remove emojis from headers.
